.github/workflows/kong3.0_compatibility.yaml

```diff
@@ -0,0 +1,62 @@
+name: 'Kong 3.0 compatibility'
+
+on:
+  schedule:
+    - cron: '30 5 * * *'
+  workflow_dispatch: {}
+
+jobs:
+  integration-tests:
+    environment: "Configure ci"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        database-mode:
+          - 'dbless'
+          - 'postgres'
+        kong-edition:
+          - 'OSS'
+          - 'enterprise'
+    steps:
+    - name: setup golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: '^1.19'
+
+    - name: cache go modules
+      uses: actions/cache@v3
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-build-codegen-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-build-codegen-
+
+    - name: checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: setup test kong image for OSS edition
+      if: ${{ matrix.kong-edition }} == 'OSS'
+      run: TEST_KONG_IMAGE=kong/kong;TEST_KONG_TAG=3.0.0-alpha.14-alpine
+
+    - name: setup test kong image and env for enterprise edition
+      if: ${{ matrix.kong-edition }} == 'enterprise'
+      run: TEST_KONG_IMAGE=kong/kong-gateway-internal;TEST_KONG_TAG=3.0.x.x-alpine;TEST_KONG_ENTERPRISE="true"
+
+    - name: setup database mode for dbless
+      if: ${{ matrix.database-mode }} == 'dbless'
+      run: TEST_DATABASE_MODE=off
+
+    - name: setup database mode for postgres
+      if: ${{ matrix.database-mode }} == 'postgres'
+      run: TEST_DATABASE_MODE=postgres
+    
+    - name: run integration tests
+      run: make _test.integration
+      env:
+        KONG_LICENSE_DATA: ${{ steps.license.outputs.license }}
+        TEST_KONG_PULL_USERNAME: ${{ secrets.GHA_DOCKERHUB_PULL_USER }}
+        TEST_KONG_PULL_PASSWORD: ${{ secrets.GHA_KONG_ORG_DOCKERHUB_PUBLIC_TOKEN }}
+
+  # e2e-tests: ## TODO: blocked by loading kong image in manifests.
```

Comment:
Do you mind adding the link to the issue for this in here?

Reply:
updated